### PR TITLE
added a .gitpod.yml

### LINKED ISSUE #611: Full support for Gitpod
## Description
Similar to the issue open for supporting GitHub CodeSpaces. When you try to use Gitpod for sir-lancebot, you have to install lots of things manually.

## Reasoning
Quicker development times online as there is no need to manually install dependencies.

## Proposed Implementation
Creating a .gitpod.yml file, described [here](https://www.gitpod.io/docs/configuration/)

## Would you like to implement this yourself?

- [x] I'd like to implement this feature myself
- [ ] Anyone can implement this feature

### PR DESCRIPTION
**Pull request by Kronifer**
## Relevant Issues

Closes #611 

## Description

Adds a `.gitpod.yml`  file that automatically installs Python 3.9.6, poetry and installs poetry dependencies.

## Did you:

- [x] Join the [**Python Discord Community**](https://discord.gg/python)?
- [x] Read all the comments in this template?
- [x] Ensure there is an issue open, or link relevant discord discussions?
- [x] Read and agree to the [contributing guidelines](https://pythondiscord.com/pages/contributing/contributing-guidelines/)?

### CODE REVIEW

**Comment by onerandomusername** [comment]
Hi, given the changes that this makes, I believe policy is to squash them into one commit. 
View https://pythondiscord.notion.site/Squashing-a-Pull-Request-191e8efa2fbb4d72840fd17ca37cf7fa for how to do that.

**Comment by Kronifer** [comment]
> Hi, given the changes that this makes, I believe policy is to squash them into one commit.
> View https://pythondiscord.notion.site/Squashing-a-Pull-Request-191e8efa2fbb4d72840fd17ca37cf7fa for how to do that.

squashed

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+tasks:
+  - name: Initialize Poetry Environment
+    before: pyenv install 3.9.6 && pyenv global 3.9.6
+    env:
+      PIP_USER: false
```
Comment (Bluenix2) [inline .gitpod.yml:5]: Is this for Gitpod itself?

File: .gitpod.yml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+tasks:
+  - name: Initialize Poetry Environment
```
Comment (Bluenix2) [inline .gitpod.yml:2]: This will become the name of the terminal itself, is it suitable for that though?

**Review by Bluenix2 — COMMENTED** [review]
I am looking at several tutorials and Gitpods reference itself, and this looks pretty good. Just have a few questions

**Comment by onerandomusername** [comment]
I have a comment on the extensions installed, too. I think this should be
installing the python and pylance extension, at minimum.

File: .gitpod.yml (hunk 1 above)
Comment (Kronifer) [reply .gitpod.yml:5]: This is so poetry can install deps, user-site packages aren't supported

File: .gitpod.yml (hunk 2 above)
Comment (Kronifer) [reply .gitpod.yml:2]: I think so

**Comment by Kronifer** [comment]
@onerandomusername I'll work on that later

File: .gitpod.yml (hunk 2 above)
Comment (Bluenix2) [reply .gitpod.yml:2]: I'd suggest something more like just a bare:
```suggestion
  - name: Python Environment
```

Comment (Kronifer) [reply .gitpod.yml:2]: I'll do this, but to clarify this only sets this as the terminal name while the commands are running

**Review by Bluenix2 — APPROVED** [review]
I'll make it clear that I do not have Gitpod and cannot test this, but this all looks good according to all documentation I could find.

File: .gitpod.yml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+tasks:
+  - name: "Python Environment"
+    before: "pyenv install 3.9.6 && pyenv global 3.9.6"
+    init: "pip install poetry && export PIP_USER=false"
+    command: "poetry install && poetry run pre-commit install"
+
+vscode:
+  extensions:
+    - "ms-python.python"
```
Comment (Bluenix2) [inline .gitpod.yml:9]: Please add a trailing new line

**Review by Bluenix2 — APPROVED** [review]
Looks good according to all documentation I could find.

**Review by jchristgit — APPROVED** [review]
This looks good, thank you.

**Review by ChrisLovering — APPROVED** [review]

**Pull request merged**